modules/web_framework.rb

```diff
@@ -0,0 +1,107 @@
+
+require 'selenium-webdriver'
+
+class Web_framework
+  @url = nil
+  def initialize()
+    Selenium::WebDriver::Chrome::Service.driver_path=ENV['driver']
+    options = Selenium::WebDriver::Chrome::Options.new
+        options.add_argument('--headless')
+        options.add_argument('--no-sandbox')
+        options.add_argument('--disable-dev-shm-usage')
+        options.add_argument('--window-size=1920,1080')
+        @driver = Selenium::WebDriver.for :chrome ,options: options
+  end
+  def get_webpage(url)
+    @url = url
+    @driver.get(url)
+  end
+  def click_element(element)
+    begin
+      return element.click
+    rescue
+      return "null element"
+    end
+
+  end
+  def find_elements(element)
+    begin
+      return @driver.find_elements(element)
+    rescue
+      return nil
+    end
+  end
+
+  def find_element(element)
+    begin
+      return @driver.find_element(element)
+    rescue
+      return nil
+    end
+  end
+
+  def wait(time)
+    if(time.is_a?Integer)
+      sleep(time)
+    else
+      return "enter a number"
+    end
```

Comment:
Instead of implementing `sleep`, you could implement the explicit/ implicit selenium wait.